theworkoutinfo/Managers/AuthenticationManager.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  AuthenticationManager.swift
+//  theworkoutinfo
+//
+//  Created by Gonzalo Barrios on 3/21/18.
+//  Copyright © 2018 Gonzalo Barrios. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+
+enum LoginResult {
+    case success
+    case wrongPassword
+    case wrongEmail
+}
+
+class AuthenticationManager {
+    
+    typealias AuthenticationCredentials = (email: String, password: String)
+    
+    static let sharedInstance = AuthenticationManager()
+    
+    var trySigning: AnyObserver<AuthenticationCredentials>
+    var signInResult: Observable<LoginResult>
+    
+    init() {
+        let _trySigning = PublishSubject<AuthenticationCredentials>()
+        trySigning = _trySigning.asObserver()
+        signInResult = _trySigning.asObservable().map { (authCredentials) -> LoginResult in
+            let userCredentials = AuthenticationHelper.getUserCredentials()
+            let credentialsOk = (authCredentials.email == userCredentials.email && authCredentials.password == userCredentials.password)
+            switch credentialsOk {
+            case true:
+                return LoginResult.success
+            case false:
+                let result: LoginResult
+                (authCredentials.email == userCredentials.email) ? (result = LoginResult.wrongPassword) : (result = LoginResult.wrongEmail)
```

Comment:
It's not secure to give the user the information about what field failed. We can discuss this